packages/api/src/services/repository/repository.service.spec.ts

```diff
@@ -16,22 +16,36 @@ describe('RepositoryService', () => {
     repositoryService = new RepositoryService();
   });
 
-  describe('getRepositories', () => {
-    test('should return repositories', async () => {
-      // Mock the response from RepositoryRestService.
-      const rawRepository = {id: 'repo-id', title: 'repo-title', location: ''};
-      const mockResponse = {'': [rawRepository]};
-      mockRepositoryRestService.getRepositories.mockResolvedValue(mockResponse);
-      const expectedResult = new RepositoryList([new Repository(rawRepository as Repository)]);
-
-      // When the service is called to fetch all repositories,
-      const repositories = await repositoryService.getRepositories();
-
-      // Then I expect it to return an instance of RepositoryList,
-      expect(repositories).toBeInstanceOf(RepositoryList);
-      // with all repositories included.
-      expect(repositories).toEqual(expectedResult);
-    });
+  test('should return repositories', async () => {
+    // Mock the response from RepositoryRestService.
+    const rawRepository = {id: 'repo-id', title: 'repo-title', location: ''};
+    const mockResponse = {'': [rawRepository]};
+    mockRepositoryRestService.getRepositories.mockResolvedValue(mockResponse as unknown as Record<string, []>);
+    const expectedResult = new RepositoryList([new Repository(rawRepository as Repository)]);
+
+    // When the service is called to fetch all repositories,
+    const repositories = await repositoryService.getRepositories();
+
+    // Then I expect it to return an instance of RepositoryList,
+    expect(repositories).toBeInstanceOf(RepositoryList);
+    // with all repositories included.
+    expect(repositories).toEqual(expectedResult);
   });
 
+  test('should return an instance of RepositorySizeInfo', async () => {
+    const rawRepositorySizeInfo = {
+      inferred: 2,
+      total: 3,
+      explicit: 1
+    };
+    mockRepositoryRestService.getRepositorySizeInfo.mockResolvedValue(rawRepositorySizeInfo as unknown as RepositorySizeInfo);
```

Comment:
I don't like how these tests are going! You are mocking the wrong part and with wrong data. And then, the inocation below is also wrong which might hide issues.
Lets discuss this tomorrow!

Reply:
Refactored